WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift

```diff
@@ -1,5 +1,38 @@
 import UIKit
 
+enum StatsPeriodType: Int, FilterTabBarItem {
+    case insights = 0
+    case days
+    case weeks
+    case months
+    case years
+
+    // This is public as it is needed by FilterTabBarItem.
+    var title: String {
+        switch self {
+        case .insights: return NSLocalizedString("Insights", comment: "Title of Insights stats filter.")
+        case .days: return NSLocalizedString("Days", comment: "Title of Days stats filter.")
+        case .weeks: return NSLocalizedString("Weeks", comment: "Title of Weeks stats filter.")
+        case .months: return NSLocalizedString("Months", comment: "Title of Months stats filter.")
+        case .years: return NSLocalizedString("Years", comment: "Title of Years stats filter.")
+        }
+    }
+}
+
+fileprivate extension StatsPeriodType {
+    static let allPeriods = [StatsPeriodType.insights, .days, .weeks, .months, .years]
```

Comment:
If `StatsPeriodType` conforms to [`CaseIterable`](https://developer.apple.com/documentation/swift/caseiterable) then you can just use `StatsPeriodType.allCases` to get a collection of all its cases